### ISSUE TITLE
How to do search using regular expression

### ISSUE THREAD

**Issue by nitinjain1105** [k1complete/qlc#4]
Hi @k1complete 

In condition list, I have seen equality, inequality, greater than or less than but do you also have search based on regular expression?

Suppose the actual value of K is "Trying to search" but in my query I dont want to provide the whole search string, instead I want to search based on only some part of the string, for example Item = "Trying to"

Qlc.q("[X || X = [K,V] <- L, K == Item]", L: list, Item: "Trying to")

In other words:
if K = "Trying to search"
and Item = "Trying to"

can we still compare?
K == Item

Thank you so much in advance.

**Comment by k1complete** [comment]
Hi, thank you your report.
You may use passing any function to Qlc argument:

  iex>list = [["Trying to search", 1], ["Search Trying", 2]]
  [["Trying to search", 1], ["Search Trying to", 2]]
 iex> Qlc.q("[X || X = [K,V] <- L, F(K)]", L: list, F: &(Regex.match?(~r/Trying to/, &1)))|>Qlc.e()
  [["Trying to search", 1]]

cf) The "&( Regex.match?(~r/Trying to/, &1 )" is equivalence to "fn(x) -> Regex.match?(~r/Trying to/, x) end".

Regards.

**Comment by nitinjain1105** [comment]
Thanks @k1complete
